src/libs/actions/Policy.js

```diff
@@ -144,51 +144,46 @@ function createAndNavigate(name = '') {
  * and we also don't have to wait for full policies to load before navigating to the new policy.
  */
 function getPolicyList() {
-    const policyCollection = {};
     API.GetPolicySummaryList()
         .then((data) => {
             if (data.jsonCode === 200) {
-                const policyDataToStore = _.reduce(data.policySummaryList, (memo, policy) => ({
+                const policyCollection = _.reduce(data.policySummaryList, (memo, policy) => ({
                     ...memo,
                     [`${ONYXKEYS.COLLECTION.POLICY}${policy.id}`]: getSimplifiedPolicyObject(policy),
                 }), {});
-                updateAllPolicies(policyDataToStore);
-            }
 
-            if (shouldCreateNewPolicy) {
-                Navigation.dismissModal();
-                Navigation.navigate(newPolicyID ? ROUTES.getWorkspaceInitialRoute(newPolicyID) : ROUTES.HOME);
+                if (!_.isEmpty(policyCollection)) {
+                    updateAllPolicies(policyCollection);
+                }
             }
         });
 }
 
+function createAndGetPolicyList() {
+    let newPolicyID;
+    create()
+        .then((policyID) => {
+            newPolicyID = policyID;
+            return getPolicyList();
+        })
+        .then(() => navigateToPolicy(newPolicyID));
+}
+
 /**
  * @param {String} policyID
  */
 function loadFullPolicy(policyID) {
-    API.GetPolicyList([policyID])
+    API.GetFullPolicy(policyID)
         .then((data) => {
             if (data.jsonCode === 200) {
-                const policy = data.policyList[0] || {};
+                const policy = lodashGet(data, 'policyList[0]', {});
```

Comment:
TIL: Did not know this worked, thought you had to do `lodashGet(data, ['policyList', 0], {});`